modules/programs/firefox.nix

```diff
@@ -99,17 +190,34 @@ in
       in
         [ (wrapper cfg.package { }) ];
 
-    home.file.".mozilla/${extensionPath}" = mkIf (cfg.extensions != []) (
-      let
-        extensionsEnv = pkgs.buildEnv {
-          name = "hm-firefox-extensions";
-          paths = cfg.extensions;
-        };
-      in
-        {
-          source = "${extensionsEnv}/share/mozilla/${extensionPath}";
-          recursive = true;
-        }
-    );
+    home.file = mkMerge ([{
+      ".mozilla/${extensionPath}" = mkIf (cfg.extensions != []) (
+        let
+          extensionsEnv = pkgs.buildEnv {
+            name = "hm-firefox-extensions";
+            paths = cfg.extensions;
+          };
+        in
+          {
+            source = "${extensionsEnv}/share/mozilla/${extensionPath}";
+            recursive = true;
+          }
+      );
+
+      ".mozilla/firefox/profiles.ini" = mkIf (cfg.profiles != {}) {
+        text = profilesINI;
+        recursive = true;
```

Comment:
No need for `recursive = true` for files, it will only have an effect for when the `source` attribute is set to a directory. Can remove it here and for the other two cases below.

Reply:
Thx, didn't knew that

Reply:
Fixed